File: Assets/Multiplayer/Scripts/Backend/Dto/RoomResponse.cs
using System;

namespace Murang.Multiplayer.Backend.Dto
{
    /// <summary>
    /// Spring → Client 룸 응답. <c>POST /api/v1/rooms</c> 생성 응답과
    /// <c>GET /api/v1/rooms/{id}</c> 조회 응답이 같은 모양을 공유한다.
    ///
    /// <para>ready callback 도달 전에는 <c>taskPublicIp</c>/<c>gamePort</c>/<c>readyAt</c>
    /// 가 null 이며 <c>status</c> 는 보통 <c>PROVISIONING</c> 또는 <c>SERVER_STARTING</c>.
    /// 클라이언트는 <c>status == READY</c> 가 될 때까지 GET 으로 폴링한다.</para>
    /// </summary>
    [Serializable]
    public sealed class RoomResponse
    {
        public long roomId;
        public string photonSessionName;
        public int maxPlayers;
        public bool locked;
        public string status;
        public string taskPublicIp;
        public int gamePort;
        public string createdAt;
        public string readyAt;

        public bool IsReady => string.Equals(status, "READY", StringComparison.Ordinal);

        public bool IsTerminal =>
            string.Equals(status, "TERMINATED", StringComparison.Ordinal)
            || string.Equals(status, "FAILED", StringComparison.Ordinal)
            || string.Equals(status, "UNHEALTHY", StringComparison.Ordinal)
            || string.Equals(status, "TERMINATING", StringComparison.Ordinal);
    }
}

File: backend/src/main/java/com/murang/room/controller/dto/RoomResponse.java
package com.murang.room.controller.dto;

import com.murang.room.domain.RoomServerInstanceStatus;
import com.murang.room.manager.RoomServerSnapshot;
import java.time.Instant;

/**
 * Spring → Client 룸 상태 응답 body. POST /api/v1/rooms 생성 응답과
 * GET /api/v1/rooms/{id} 조회 응답이 같은 모양을 공유한다.
 *
 * <p>{@code taskPublicIp}/{@code gamePort} 는 ready callback 도달 전까지
 * null. {@code readyAt} 도 마찬가지. 클라이언트는 status=READY 와
 * taskPublicIp/gamePort 모두 채워진 시점에 Photon 합류를 시도한다.
 */
public record RoomResponse(
        Long roomId,
        String photonSessionName,
        int maxPlayers,
        boolean locked,
        RoomServerInstanceStatus status,
        String taskPublicIp,
        Integer gamePort,
        Instant createdAt,
        Instant readyAt
) {

    public static RoomResponse of(RoomServerSnapshot snapshot) {
        return new RoomResponse(
                snapshot.roomId(),
                snapshot.photonSessionName(),
                snapshot.maxPlayers(),
                snapshot.locked(),
                snapshot.status(),
                snapshot.taskPublicIp(),
                snapshot.gamePort(),
                snapshot.createdAt(),
                snapshot.readyAt()
        );
    }
}
